### PR TITLE
Story4

### PR DESCRIPTION
**Pull request by vivhayles**
Pls merge

### CODE REVIEW

File: app/Http/Controllers/BookingAPIController.php [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+{
+    public function index()
+    {
+        $currentAndFutureBookings = Booking::with('room:id,name')
+            ->where('end', '>', now())
+            ->get()?->makeHidden(['updated_at', 'room_id'])
+            ->map(function ($booking) {
+                $booking->room->makeHidden(['id']);
+                return $booking;
+            })
```
Comment (ashleycoles) [inline app/Http/Controllers/BookingAPIController.php]: I believe the ID actually needs to stay - the front-end does (I think) use that information. Even if it doesn't, the rooms do have both id and name in the docs, so you should stick with that

Cool use of map tho, never considered using it for stuff like this.

File: app/Models/Booking.php [hunk 2]
```diff
@@ -2,9 +2,15 @@
...
 
+use Illuminate\Database\Eloquent\Factories\HasFactory;
 use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\Relations\BelongsTo;
 
 class Booking extends Model
 {
-    //
+    use HasFactory;
+    public function room(): BelongsTo
```
Comment (ashleycoles) [inline app/Models/Booking.php:13]: Space above method plz

File: app/Models/Room.php [hunk 3]
```diff
@@ -14,4 +15,8 @@ public function type(): BelongsTo
     {
         return $this->belongsTo(Type::class);
     }
+    public function booking(): HasMany
```
Comment (ashleycoles) [inline app/Models/Room.php]: Space above this method

**Review by ashleycoles — CHANGES_REQUESTED** [review]

**Comment by Anerkiki** [comment]
Resolved comments, not sure how to do/if need to do Changes requested

**Comment by Anerkiki** [comment]
I think I meant conflicts

**Review by ashleycoles — APPROVED** [review]

**Pull request merged**